napari/utils/colormaps/colormap.py

```diff
@@ -534,7 +534,12 @@ def _get_typed_dict_mapping(self, data_dtype: np.dtype) -> 'typed.Dict':
     @cached_property
     def _array_map(self):
         """Create an array to map labels to texture values of smaller dtype."""
-        max_value = max(x for x in self.color_dict if x is not None)
+
+        max_value = max(
+            (abs(x) for x in self.color_dict if x is not None), default=0
+        )
+        if any(x < 0 for x in self.color_dict if x is not None):
```

Comment:
We really need to deprecate None and use defaultdict I think... (next PR) Anyway, like below this isn't tested — would you like it to be?

Reply:
It is tested, but not with code coverage (in pip job). I will add additional tests

Reply:
We do not want to use default dict globally. As defaultdict is adding elements when they are missed. So size is increasing.

Reply:
The option may be to add an additional property to `DirectLAbelColormap` to store default color outside color dict

Reply:
Good point @Czaki, I like this option.